### ISSUE TITLE
Annotation listing not working

### ISSUE THREAD

**Issue by uskudarli** [swe574-group2/swe574#73]
I have created 2 annotations on https://medium.com/@lancegpowelljr/social-vr-and-the-2016-presidential-debate-68a78e73b3d9#.g22g3d252 page. The first one shows (image), the second one doesn't show.

**Comment by penumbram-zz** [comment]
@uskudarli beginning to work on it shortly

**Comment by penumbram** [comment]
@uskudarli [this commit](https://github.com/swe574-group2/swe574/commit/c06839519bde6e07442b6ce4f9f5803b284c09bd) fixes the bug described above, awaiting confirmation.

On a side note, to make goals clear here, shall we set a deadline for these issues? @uskudarli

**Comment by penumbram-zz** [comment]
Sorry for the possible confusion, I labeled this one as fixed, seems I mixed it up with the other bug I've been dealing with. This one is still in progress. 

On a side note, to make goals clear here, shall we set a deadline for these issues? @uskudarli Just to make our priorities and time management more effective.

**Comment by penumbram-zz** [comment]
I've investigated the issue and it seems like the url you posted https://medium.com/@lancegpowelljr/social-vr-and-the-2016-presidential-debate-68a78e73b3d9#.g22g3d252

changes everytime its reloaded, specifically the part after the "#".
Now there are different ways it can go down, either we add medium.com as a special case and check its urls for hashtag characters, and ignore the parts after hashtag (which sounds un-"software engineer"ly) or we research the url pattern the website is using and if we are lucky we find the pattern, and once we know how to recognise urls like this, it'll be solved for all the urls. But from my front-end experience I know it's possible for the client to change the url however it wants, and this issue could be more of a general issue for the annotation tools. I certainly am not sure.

The first solution is bad but it'll work like a charm, the second solution is better but it might not be possible, ever. We're gonna discuss this among ourselves and present a solution, meanwhile we are open to your input and guidance @uskudarli .

**Comment by okanmenevseoglu** [comment]
@uskudarli I agree with @penumbram . The link you are trying has a changing source (the page url). They do persist on the database as I have looked but you can't get them because of the source changes. We will look into this situation, meanwhile you can try static links for the best result.

**Comment by uskudarli** [comment]
OK.

I just got home. After I warm up a bit. I will give it another go.

Thanks for your speedy and clear answers.

More later
-s

**Comment by uskudarli** [comment]
Actually, it is nice... The challenges of annotation creation apps are many.

Wonder why a blog URL changes? I'll take a look at it as well.

**Comment by uskudarli** [comment]
There is also the issue of the need to clear highlights. So easy to forget. 

Now I have 2 annotations on http://mitan.in/bcp/raika/

**Comment by okanmenevseoglu** [comment]
@uskudarli I guess this, #76  and #74 are solved now and you can annotate on non-dynamic links?

**Comment by penumbram-zz** [comment]
@uskudarli I fixed it with [this](https://github.com/swe574-group2/swe574/commit/a95cb79f07eacb374864df06f03cea4fc6e3dba5) as I described in this thread before (the first & dirty way). The funny thing is, Google Chrome doesn't show the part after "#." in the url box, so I'm guessing it wouldn't cause a probem in Chrome, but firefox does show the whole url. 

Judging by Chrome's behaviour, I'm guessing this "#." at the end is a common way to separate and distinguish urls. So since Chrome also clips the part after "#.", I don't think the fix is as dirty as I thought it would be in the first place.

**Comment by uskudarli** [comment]
Actually I have been testing with Firefox. 

Well, interesting. Thanks for the feedback. But, I can't say that I understand the need for this behavior on blogs. Did you understand it? This is beyond the scope of our project. But will be good to know.

**Comment by penumbram-zz** [comment]
Yes, you should, we don't (officially) support Chrome anyway. :)

I don't think it's related to blogs, but I think some websites are trying to differentiate their share urls, probably for tracking links and/or affiliate marketing purposes. Different links would help them track the traffic via these different urls, in return it would make it possible to count how many people used a specific link to visit their website, which could be used to decide how much they should pay each of their affiliates (identified by these different links). I'm aware of such practices, though for this case I'm not 100% sure. Just trying to connect the dots.

**Comment by uskudarli** [comment]
I am on Firefox anyway. Not testing on Chrome.

OK got it. I had not installed the most recent plugin.

It is pretty chaotic on my end just now (parallel evaluation several projects, don't recommend it! :) )

Thanks for your speedy and informative responses. Much appreciated.

[for me this issue is closed]

**Comment by okanmenevseoglu** [comment]
@uskudarli I have investigated the different URL's on the same content and I have found out that the page is not exactly "static". Each link has suggesting different combination of 3 different articles that we can click on and go to at the end of the page. These links may refer to these different combinations. The site owners may analyze the conversion rates for each dynamic link to decide which combination has the best rate. These type of things can be useful for A/B testings or when you are simply trying to increase your visitors.
